uwblueprint/social-diversity-for-children
# feat(class): My classes page with multiple/single children workflow

### Notion ticket link



[My Classes Page](https://www.notion.so/uwblueprintexecs/My-Classes-Page-bc2bbf5da6a54dc0ae2fd82958ff8b5d)

Parent View:

https://user-images.githubusercontent.com/44826218/135002195-e02bff7a-7c07-49b4-b991-a6c88c0172e0.mp4

Volunteer View:

https://user-images.githubusercontent.com/44826218/135723929-605a9940-eb52-4a64-84db-cc309993375f.mp4



### Implementation description

- Implement Figma My Classes page for case of multiple and single children
- Unregister button which calls delete REST API for parent registration
- Stud join the class

Note: This PR does not include waitlist features.



### Steps to test

Parent View:
1. Create parent registrations with at least two different children across 2 classes. There should be at least three parent registration
2. Use the tabs and navigate between children. Expected: For a certain tab, we only show enrollment records for that children
3. Click the hamburger icon and unregister children. Expected: The record disappears from my classes page
3b. Step 3 should also work in any specific tab
4. Repeat until there are only records for 1 child. Expected: The tabs disappear and the page shifts to the single children view

Volunteer View:
1. same as parent view, but with a single volunteer user
2. the register drop down should only say "unregister"
3. the register works by unregistering the volunteer from the class



### What should reviewers focus on?

- Design: Look and feel of my classes page
- Design: Flow of multiple/single page
 - Unregister prompt for single page
- Dev: New API calls for delete reg
- Dev: New components
- Dev: MeQuery
- Dev: Volunteer/Parent flow
- Dev: Loading component

### Checklist
